std/algorithm/iteration.d

```diff
@@ -2706,21 +2706,24 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR))
 
         private enum popFrontEmptyElements = q{
             // Skip over empty subranges.
-            if (_items.empty) goto end;
-            while (_items.front.empty)
+            while (!_items.empty && _items.front.empty)
             {
                 _items.popFront();
-                if (_items.empty) goto end;
             }
-            // We cannot export .save method unless we ensure subranges are not
-            // consumed when a .save'd copy of ourselves is iterated over. So
-            // we need to .save each subrange we traverse.
-            static if (isForwardRange!RoR && isForwardRange!(ElementType!RoR))
-                _current = _items.front.save;
+            if (!_items.empty)
```

Comment:
This is sub-optimal as it will be executed one-more time than strictly necessary.

Reply:
It should be faster to do sth. this:

```
while (!_items.empty) {
   if (!_items.front.empty) {
      _items.popFront();
   } else {
     // <- static if code here
     break;
   }
}
```

Reply:
I still need the outside `if (!_items.empty) { } else { }` block, or the compiler will complain with `Error: one path skips field _current`.

This is because, in the case of an empty ROR, `_current` won't be initialised.